### ISSUE TITLE
CODEOWNERS is not enforced correctly

### ISSUE THREAD

**Issue by hasheddan** [crossplane/crossplane#3724]
### What happened?

I opened https://github.com/crossplane/crossplane/pull/3719, which modified `README.md`, which is owned by the `steering-committee` group. That group was correctly requested for review, but when a member of `crossplane-reviewers` approved, I was permitted to merge as a maintainer.

![Screenshot from 2023-02-03 11-08-30](https://user-images.githubusercontent.com/31777345/216651559-380c1239-2c93-431a-8a52-ba38101c2542.png)

![Screenshot from 2023-02-03 11-11-03](https://user-images.githubusercontent.com/31777345/216651691-861672e6-5af2-450a-998e-74a3e096bb4b.png)

I believe this may be due to our settings not [requiring the review](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#:~:text=Optionally%2C%20you%20can%20choose%20to%20require%20reviews%20from%20code%20owners.%20If%20you%20do%2C%20any%20pull%20request%20that%20affects%20code%20with%20a%20code%20owner%20must%20be%20approved%20by%20that%20code%20owner%20before%20the%20pull%20request%20can%20be%20merged%20into%20the%20protected%20branch.) from the most specific code owner:

> Optionally, you can choose to require reviews from code owners. If you do, any pull request that affects code with a code owner must be approved by that code owner before the pull request can be merged into the protected branch.

### How can we reproduce it?

Open a PR that should require `steering-committee` approval and have a reviewer approve it.

### What environment did it happen in?
Crossplane version: 

`master`

**Comment by jbw976** [comment]
Good find @hasheddan!! I have enabled the following on the branch protection rule for master:

> Require review from Code Owners
Require an approved review in pull requests including files with a designated code owner.

Now on your PR #3719, I see:

> Code owner review required
Waiting on code owner review from crossplane/steering-committee. 

Looks like we now have the intended behavior of the code owners assignments - I'll make the same thing to the `release-*` branch protection rule.   Let me know if I've misunderstood the intended behavior and/or if any more tweaks are needed! 🙇
